source/secure_tunneling.c

```diff
@@ -297,174 +330,64 @@ static void s_process_iot_st_msg(struct aws_secure_tunnel *secure_tunnel, struct
 static void s_process_received_data(struct aws_secure_tunnel *secure_tunnel) {
```

Comment:
Makes sense. Changed it. It will close down the websocket now on a decode error and kick off the reconnect process to make itself available to fresh data.